lib/CStarToC11.ml

```diff
@@ -641,16 +647,30 @@ and mk_stmt m (stmt: stmt): C.stmt list =
       (* In the case where this is a buffer creation in the C* meaning, then we
        * declare a fixed-length array; this is an "upcast" from pointer type to
        * array type, in the C sense. *)
-      let t, init, size = ensure_array m binder.typ rhs in
-      (* TODO: make ensure_array return both *)
-      let n_elements = match size with
-        | C.Op2 (K.Mult, outer_size, _) -> outer_size
-        | _ -> failwith "impossible, see ensure_array"
+      let t, init, n_elements, e_size = ensure_array m binder.typ rhs in
+
+      let is_constant = match n_elements with Constant _ -> true | _ -> false in
```

Comment:
you could have a slightly better criterion here, and include variables, or arithmetic expressions -- this is going to generate a diff in HACL's bignums otherwise (which use VLAs in a limited number of places)

Reply:
yeah but then you'd have to go in depth to make sure that no `Call` nodes are hidden in there... might be a little bit of engineering -- I'm fine either with a simple criterion (`let rec is_simple_expression`) or a more general one (`object inherit [_] reduce_cstar method! visit_BufCreate _ _ _ _ = false method! visit_BufCreateL _ _ _ _ = false method! visit_Call _ _ _ = false etc.`)

Reply:
Pushed an update (without objects...)